docs/contributor/image_mode.md

```diff
@@ -0,0 +1,217 @@
+# Image Mode for MicroShift
+
+Image mode is a new approach to operating system deployment that lets users build,
+deploy, and manage Red Hat Enterprise Linux as a bootable container (`bootc`) image.
+MicroShift build and deployment procedures can utilize bootable containers to
+benefit from this technology.
+
+This document demonstrates how to:
+* Create a bootable container image and boot a RHEL operating system from this
+image using `podman`
+* Store a `bootc` image in a remote registry and use it for installing a new
+virtual machine
+* Upgrade an existing `rpm-ostree` system created with `composer-cli` tools to
+use `bootc` images
+
+> See [Image mode for Red Hat Enterprise Linux](https://developers.redhat.com/products/rhel-image-mode/overview)
+for more information.
+
+The procedures described below require the following setup:
+* A `physical hypervisor host` with the [libvirt](https://libvirt.org/) virtualization
+platform for starting virtual machines that run RHEL OS containing MicroShift binaries
+* A `development virtual machine` set up according to the [MicroShift Development Environment](./devenv_setup.md)
+instructions for building MicroShift `bootc` images and running containers
+* A `remote registry` (i.e. `quay.io`) for storing and accessing `bootc` images
+
+## Run MicroShift Bootc Image
+
+Log into the `development virtual machine` with the `microshift` user credentials.
+
+### Build Image
+
+Create the `Containerfile` file with the following contents.
+
+```
+FROM registry.redhat.io/rhel9/rhel-bootc:9.4
+
+RUN dnf config-manager \
+        --set-enabled rhocp-4.16-for-rhel-9-x86_64-rpms \
```

Comment:
for arch we can have use it like `rhocp-4.16-for-rhel-9-$(uname -m)-rpms` so we can build it for arm64 also without much change (like on mac M1). Also version we can take as argument with default value set to 4.16.